src/commands/base-command.js

```diff
@@ -429,6 +431,18 @@ class BaseCommand extends Command {
     const apiOpts = { ...apiUrlOpts, agent }
     const globalConfig = await getGlobalConfig()
     const { NetlifyAPI } = await jsClient
+    const configWatcher = new events.EventEmitter()
+
+    watchDebounced(configPath, {
```

Comment:
I would prefer if we do this only when `--graph` is supplied for now, and not for all commands as it can have unexpected behavior.
For example this assumes the new config is propagated to where it's used, however in some cases we only pass a subset of the config. For example:
https://github.com/netlify/cli/blob/7b6b94a2fb6bf1258b5f43aec770f768399512ad/src/commands/dev/dev.js#L288
https://github.com/netlify/cli/blob/7b6b94a2fb6bf1258b5f43aec770f768399512ad/src/commands/dev/dev.js#L302

Also we would need to check if we need to close the watcher (the one returned from `watchDebounced`), as unclosed file watchers have created issues in our tests in the past (mostly flakiness)

Reply:
One option would be to not write the updated config to the `netlify.config` object (https://github.com/netlify/cli/pull/4268/files#diff-fc2c4aaa55fa6705ff0cf8ecdb6637ee37c932821b806fb3bef0a3b72afa309aR443), and instead supply it as a parameter to the `change` event.

So `netlify.config` would be immutable, which means it shouldn't affect the functionality of existing commands, and whoever wants to subscribe to the `change` event would take the updated config from the event and update it wherever it's needed.

Reply:
Thanks @eduardoboucas, that's a better approach as it will allow us to gradually adopt this change

Reply:
thanks, first step done in 9304cb6e